lib/req/steps.ex

```diff
@@ -849,6 +857,13 @@ defmodule Req.Steps do
     end
   end
 
+  defp put_redirect_request_method(%Req.Request{method: :get} = request), do: request
```

Comment:
This isn't technically necessary but wasn't sure if it would save an unnecessary update to the struct when the method is already GET.

Reply:
I think either way is fine. I'm gonna remove the clause so we have less code around.